src/balloontoolbareditorui.js

```diff
@@ -69,7 +69,13 @@ export default class BalloonToolbarEditorUI {
 			origin: editor.editing.view,
 			originFocusTracker: this.focusTracker,
 			originKeystrokeHandler: editor.keystrokes,
-			toolbar: contextualToolbar.toolbarView
+			toolbar: contextualToolbar.toolbarView,
+			beforeFocus: () => {
```

Comment:
BTW, don't you think that `beforeFocus` callback doing `show()` is kinda odd? It's hard to read. I had to look into the code to figure out what's going on here.